trace/exemplar_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace_test
+
+import (
+	"context"
+	"testing"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+func TestTraceExemplar(t *testing.T) {
+	m := stats.Float64("measure."+t.Name(), "", stats.UnitDimensionless)
+	v := &view.View{
+		Measure:     m,
+		Aggregation: view.Distribution(0, 1, 2, 3),
+	}
+	view.Register(v)
+	ctx := context.Background()
+	ctx, span := trace.StartSpan(ctx, t.Name(), trace.WithSampler(trace.AlwaysSample()))
+	stats.Record(ctx, m.M(1.5))
+	span.End()
+
+	rows, err := view.RetrieveData(v.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(rows) == 0 {
+		t.Fatal("len(rows) = 0; want > 0")
+	}
+	dd := rows[0].Data.(*view.DistributionData)
+	if got := len(dd.Buckets); got < 3 {
+		t.Fatalf("len(buckets) = %d; want >= 2", len(dd.Buckets))
+	}
+	bucket := dd.Buckets[2]
+	if got, want := bucket.Count, int64(1); got != want {
+		t.Fatalf("bucket.Count = %d; want %d", got, want)
+	}
+	exemplar := bucket.Exemplar
+	if exemplar == nil {
+		t.Fatal("Expected exemplar")
+	}
+	if got, want := exemplar.Value, 1.5; got != want {
+		t.Fatalf("exemplar.Value = %v; got %v", got, want)
+	}
+	if _, ok := exemplar.Attachments["TraceID"]; !ok {
```

Comment:
Could we please use ExemplarTraceID and ExemplerSpanID, etc.. as the keys in these tests?

Reply:
These are the keys from the spec. I think since they map is already inside the exemplar, repeating the work "exemplar" doesn't really add anything.